src/codec/mod.rs

```diff
@@ -115,6 +121,28 @@ impl From<Response> for Bytes {
                 data.put_u16(address);
                 data.put_u16(word);
             }
+            ReadDeviceIdentification(
+                read_device_id_code,
+                conformity_level,
+                more_follows,
+                next_object_id,
+                device_id_objects,
+            ) => {
+                data.put_u8(0x0e);
+                data.put_u8(read_device_id_code);
+                data.put_u8(conformity_level);
+                data.put_u8(if more_follows { 0xff } else { 0x00 });
+                data.put_u8(next_object_id);
+                /* TODO: do not panic! */
+                data.put_u8(device_id_objects.len().try_into().unwrap());
```

Comment:
Whats the best way to return from here with error without panic?

Reply:
Are you still interested in finishing this PR? Then please try to rebase your branch onto #160 which allows this conversion to fail.

Reply:
Ping. See my comment. You could now return an error.